src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -2290,7 +2290,7 @@ def set_index(
                 label_or_series.append(key._query_compiler)
             elif isinstance(key, (np.ndarray, list, Iterator)):
                 label_or_series.append(pd.Series(key)._query_compiler)
-            elif isinstance(key, pd.Index):
+            elif isinstance(key, (pd.Index, pandas.Index)):
```

Comment:
can we not allow pandas Index here? It is also not in the type hints.

Reply:
We have to allow pandas Index there because the modin index currently returns a pandas index